From ff0438fed6358ce06a441e0f7653e201367de581 Mon Sep 17 00:00:00 2001
From: David Lam <david@thedavid.net>
Date: Sun, 29 Nov 2020 19:13:43 -0800
Subject: [PATCH] luci-app-openvpn: update encryption values

Update the list of encryption values to reflect sensible/available values
in OpenVPN 2.5.0. Certain values are long deprecated and/or insecure and
should not be made available in LuCI.

Signed-off-by: David Lam <david@thedavid.net>
---
 .../luasrc/model/cbi/openvpn-advanced.lua     | 72 ++++++-------------
 1 file changed, 21 insertions(+), 51 deletions(-)

diff --git a/applications/luci-app-openvpn/luasrc/model/cbi/openvpn-advanced.lua b/applications/luci-app-openvpn/luasrc/model/cbi/openvpn-advanced.lua
index 04934c8c11..840ddf39a8 100644
--- a/applications/luci-app-openvpn/luasrc/model/cbi/openvpn-advanced.lua
+++ b/applications/luci-app-openvpn/luasrc/model/cbi/openvpn-advanced.lua
@@ -576,48 +576,12 @@ local knownParams = {
 			"cipher",
 			{
 				"AES-128-CBC",
-				"AES-128-CFB",
-				"AES-128-CFB1",
-				"AES-128-CFB8",
-				"AES-128-GCM",
-				"AES-128-OFB",
 				"AES-192-CBC",
-				"AES-192-CFB",
-				"AES-192-CFB1",
-				"AES-192-CFB8",
-				"AES-192-GCM",
-				"AES-192-OFB",
 				"AES-256-CBC",
-				"AES-256-CFB",
-				"AES-256-CFB1",
-				"AES-256-CFB8",
+				"AES-128-GCM",
+				"AES-192-GCM",
 				"AES-256-GCM",
-				"AES-256-OFB",
-				"BF-CBC",
-				"BF-CFB",
-				"BF-OFB",
-				"CAST5-CBC",
-				"CAST5-CFB",
-				"CAST5-OFB",
-				"DES-CBC",
-				"DES-CFB",
-				"DES-CFB1",
-				"DES-CFB8",
-				"DES-EDE-CBC",
-				"DES-EDE-CFB",
-				"DES-EDE-OFB",
-				"DES-EDE3-CBC",
-				"DES-EDE3-CFB",
-				"DES-EDE3-CFB1",
-				"DES-EDE3-CFB8",
-				"DES-EDE3-OFB",
-				"DES-OFB",
-				"DESX-CBC",
-				"RC2-40-CBC",
-				"RC2-64-CBC",
-				"RC2-CBC",
-				"RC2-CFB",
-				"RC2-OFB"
+				"CHACHA20-POLY1305"
 			},
 			translate("Encryption cipher for packets") },
 	-- parse
@@ -679,27 +643,33 @@ local knownParams = {
 		{ DynamicList,
 			"tls_cipher",
 			{
+				"TLS-ECDHE-RSA-WITH-CHACHA20-POLY1305-SHA256",
+				"TLS-ECDHE-ECDSA-WITH-CHACHA20-POLY1305-SHA256",
+				"TLS-DHE-RSA-WITH-CHACHA20-POLY1305-SHA256",
 				"TLS-ECDHE-ECDSA-WITH-AES-256-GCM-SHA384",
 				"TLS-ECDHE-RSA-WITH-AES-256-GCM-SHA384",
 				"TLS-DHE-RSA-WITH-AES-256-GCM-SHA384",
-				"TLS-ECDHE-ECDSA-WITH-CHACHA20-POLY1305-SHA256",
-				"TLS-ECDHE-RSA-WITH-CHACHA20-POLY1305-SHA256",
-				"TLS-DHE-RSA-WITH-CHACHA20-POLY1305-SHA256",
-				"TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256",
-				"TLS-ECDHE-RSA-WITH-AES-128-GCM-SHA256",
-				"TLS-DHE-RSA-WITH-AES-128-GCM-SHA256",
 				"TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA384",
 				"TLS-ECDHE-RSA-WITH-AES-256-CBC-SHA384",
 				"TLS-DHE-RSA-WITH-AES-256-CBC-SHA256",
-				"TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA256",
-				"TLS-ECDHE-RSA-WITH-AES-128-CBC-SHA256",
-				"TLS-DHE-RSA-WITH-AES-128-CBC-SHA256",
 				"TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA",
 				"TLS-ECDHE-RSA-WITH-AES-256-CBC-SHA",
 				"TLS-DHE-RSA-WITH-AES-256-CBC-SHA",
+				"TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256",
+				"TLS-ECDHE-RSA-WITH-AES-128-GCM-SHA256",
+				"TLS-DHE-RSA-WITH-AES-128-GCM-SHA256",
+				"TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA256",
+				"TLS-ECDHE-RSA-WITH-AES-128-CBC-SHA256",
+				"TLS-DHE-RSA-WITH-AES-128-CBC-SHA256",
 				"TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA",
 				"TLS-ECDHE-RSA-WITH-AES-128-CBC-SHA",
-				"TLS-DHE-RSA-WITH-AES-128-CBC-SHA"
+				"TLS-DHE-RSA-WITH-AES-128-CBC-SHA",
+				"TLS-RSA-WITH-AES-256-GCM-SHA384",
+				"TLS-RSA-WITH-AES-256-CBC-SHA256",
+				"TLS-RSA-WITH-AES-256-CBC-SHA",
+				"TLS-RSA-WITH-AES-128-GCM-SHA256",
+				"TLS-RSA-WITH-AES-128-CBC-SHA256",
+				"TLS-RSA-WITH-AES-128-CBC-SHA"
 			},
 			translate("TLS cipher") },
 		{ DynamicList,
@@ -780,7 +750,7 @@ local knownParams = {
 			translate("The lowest supported TLS version") },
 		{ Value,
 			"tls_version_max",
-			"1.2",
+			"1.3",
 			translate("The highest supported TLS version") },
 		{ ListValue,
 			"key_direction",
@@ -792,7 +762,7 @@ local knownParams = {
 			translate("This completely disables cipher negotiation") },
 		{ Value,
 			"ncp_ciphers",
-			"AES-256-GCM:AES-128-GCM",
+			"CHACHA20-POLY1305:AES-128-GCM:AES-256-GCM",
 			translate("Restrict the allowed ciphers to be negotiated") },
 	} }
 }
